### PR TITLE
[Intel MKL] Patch for fix Shape compilation issue in MKL build

### PR DESCRIPTION
**Pull request by mdfaijul**
This PR adapts a recent changes in `MakeShape` function.

### CODE REVIEW

**Comment by nhasabni** [comment]
@penpornk Can you pls take a look at this small PR that is fixing a build failure in MKL backend caused by [recent change(https://github.com/tensorflow/tensorflow/commit/271f6bb49d2140b4c1bca88391caedd1791561cf#diff-df5e1858656225cf1f668278adb47e08). Thanks in advance!

**Review by reedwm — DISMISSED** [review]

**Review by penpornk — APPROVED** [review]
Thank you for the fix!
Thank you @reedwm for the quick approve!

File: tensorflow/core/kernels/mkl_conv_grad_input_ops.cc [hunk 1]
```diff
@@ -523,7 +524,7 @@ class MklConvCustomBackpropInputOp
     // Conv[2D|3D]BackpropInputV2 supports both DT_INT32 and DT_INT64
     // output_shape tensor::MakeShape is able to handle both DT_INT32 and
     // DT_INT64 for input_tensor.
-    CHECK_EQ(this->MakeShape(input_tensor, &input_tf_shape).ok(), true);
+    CHECK_EQ(tensor::MakeShape(input_tensor, &input_tf_shape).ok(), true);
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_conv_grad_input_ops.cc]: We got a `no new check in code` error. Could you please change this to `DCHECK_EQ` instead? Thank you!
```suggestion
    DCHECK_EQ(tensor::MakeShape(input_tensor, &input_tf_shape).ok(), true);
```

**Review by penpornk — CHANGES_REQUESTED** [review]

**Review by penpornk — APPROVED** [review]
Nvm, I'll just fix this internally.

**Comment by mdfaijul** [comment]
@penpornk I have fixed it already. Thanks!

**Review by penpornk — COMMENTED** [review]
Sorry, just saw that you just fixed it. Thank you!

File: tensorflow/core/kernels/mkl_conv_grad_input_ops.cc (hunk 1 above)
Comment (mdfaijul) [reply tensorflow/core/kernels/mkl_conv_grad_input_ops.cc]: Done.

**Comment by penpornk** [comment]
Closing this since it has been merged.

**Comment by nhasabni** [comment]
Thanks for quick review @penpornk.

**Comment by nhasabni** [comment]
@penpornk Changing `CHECK_EQ` to `DCHECK_EQ` is leading to a few unit test failures in MKL build since compiler is removing that entire line all together? So we are working on PR to fix the unit test failures.

`no new check` means we cannot use any `CHECK` macro now? Can we use `TF_CHECK_OK` instead?

**Comment by penpornk** [comment]
@nhasabni Oops. Sorry about that. Yes, `DCHECK`s won't execute in opt mode. I forgot to check if the expression inside is something that needs to be executed. 

I believe `TF_CHECK_OK` is still allowed. Only `CHECK_*` are banned from new code. 

I can change `DCHECK_EQ` to `TF_CHECK_OK` right now too if that's the only fix in the PR. Let me know. Sorry again for the trouble!

**Comment by nhasabni** [comment]
@penpornk thanks for quick reply. We have started an internal unit test run to see if the change fixes the issue. We will test with TF_CHECK_OK. I will update you in an hour or so.

**Comment by nhasabni** [comment]
@penpornk We made following change, and that is passing all the internal testing. Would you like to go ahead and submit the change? Or should I submit a separate PR? Let me know.

```
     - DCHECK_EQ(tensor::MakeShape(input_tensor, &input_tf_shape).ok(), true);
     + TF_CHECK_OK(tensor::MakeShape(input_tensor, &input_tf_shape));
```

**Comment by penpornk** [comment]
@nhasabni I can submit the change. Thank you so much!

**Comment by nhasabni** [comment]
@penpornk Sounds good. Thanks!

**Comment by penpornk** [comment]
@nhasabni Fixed in this commit https://github.com/tensorflow/tensorflow/commit/dc94b3a69e8988e5db1994ce607b893f59d6f2d9

**Pull request closed without merging**